theseus/optimizer/manifold_gaussian.py

```diff
@@ -98,5 +98,42 @@ def update(
                 f"Tried to update using tensor on device: {precision.dtype} but precision "
                 f"is on device: {self.device}."
             )
+        if not torch.allclose(precision, precision.transpose(1, 2)):
+            raise ValueError("Tried to update precision with non-symmetric matrix.")
 
         self.precision = precision
+
+
+def local_gaussian(
+    variable: LieGroup,
+    gaussian: ManifoldGaussian,
+    return_mean: bool = True,
+) -> Tuple[torch.Tensor, torch.Tensor]:
+    # mean vector in the tangent space at variable
+    mean_tp = variable.local(gaussian.mean[0])
```

Comment:
Does this assume the passed ManifoldGaussian is over just one Manifold object?

Reply:
Yes, it assumes the ManifoldGaussian is over just one Manifold object which must be the same class as the variable LieGroup object.

Reply:
I've now added a check for this.